File: mssp/examples/005-before-after/src/TMS/formats/csv.js
// Renders a summary as CSV. Knows nothing of formats/text.
export const name = "formats/csv";

export function render(summary, dropped) {
  const lines = ["sensor,n,mean,min,max"];
  for (const s of summary) lines.push(`${s.sensor},${s.n},${s.mean},${s.min},${s.max}`);
  lines.push(`# ${dropped.length} row(s) dropped`);
  return lines.join("\n");
}

File: mssp/examples/007-identity-test-run/src/TMS/reports/csv.py
"""Machine-readable report. Imports nothing, takes plain values."""

NAME = "reports/csv"


def render(summary, rows):
    out = ["metric,value"]
    for key in ("matched", "differing", "only_left", "only_right", "net_difference_cents"):
        out.append(f"{key},{summary[key]}")
    out.append("")
    out.append("id,left,right,note")
    for row in rows:
        out.append(f"{row['id']},{row['left']},{row['right']},{row['note']}")
    return "\n".join(out) + "\n"
